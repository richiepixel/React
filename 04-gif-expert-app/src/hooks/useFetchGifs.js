import { useState } from 'react';

export const useFetchGifs = () => {
  const [state, setstate] = useState({
    data: [],
    loading: true,
  });

  setTimeout(() => {
    setstate(
      {
        data: [1, 2, 3, 4, 5],
        loading: false,
      },
      10000
    );
  });
  return state; //{data:[], loading: true}
};
