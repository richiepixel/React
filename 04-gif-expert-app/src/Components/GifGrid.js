import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
//import { getGifs } from '../helpers/getGifs';
// import GifGridItem from './GifGridItem';

const GifGrid = ({ category }) => {
  // const [images, setImages] = useState([]);

  const { data, loading } = useFetchGifs(); //const state = useFetchGifs();
  console.log(data);
  console.log(loading);

  // useEffect(() => {
  //   getGifs(category).then(setImages); // (imgs) => setImages(imgs)
  // }, [category]); //Cada que la categoria cambia se ejcutar√° de nuevo.

  return (
    <>
      <h3>{category}</h3>

      {loading ? 'Cargando...' : 'Data cargada'}

      {/*  <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} /> //{...img} = (img = {img})
        ))}
      </div> */}
    </>
  );
};

export default GifGrid;
