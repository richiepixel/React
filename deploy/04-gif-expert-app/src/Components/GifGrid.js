import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';

const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category); //const state = useFetchGifs();

  return (
    <>
      <h3 className="animate__animated animate__bounceInLeft">{category}</h3>

      {loading && (
        <p className="animate__animated animate__flash">Loading...</p>
      )}

      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} /> //{...img} = (img = {img})
        ))}
      </div>
    </>
  );
};

export default GifGrid;
