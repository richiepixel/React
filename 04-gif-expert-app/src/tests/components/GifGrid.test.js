import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGrid from '../../Components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebuas en <GifGrid u/>', () => {
  const category = 'Zelda';
  test('debe de mostrar el componente correctamente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar items cuando se cargan imÃ¡genes con useFetchGifs', () => {
    const gifs = [
      {
        id: 'zelda1',
        title: 'legend of zelda GIF',
        url:
          'https://media2.giphy.com/media/5MjQ87rBCRIB2/giphy.gif?cid=52d5781amo7m364mwu12b47its9vkdcwnpqx9gndzfykpgyr&rid=giphy.gif',
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
