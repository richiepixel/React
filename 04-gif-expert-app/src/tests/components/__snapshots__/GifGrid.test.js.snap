// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebuas en <GifGrid u/> debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounceInLeft"
  >
    Zelda
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebuas en <GifGrid u/> debe de mostrar items cuando se cargan im√°genes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounceInLeft"
  >
    Zelda
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="zelda1"
      key="zelda1"
      title="legend of zelda GIF"
      url="https://media2.giphy.com/media/5MjQ87rBCRIB2/giphy.gif?cid=52d5781amo7m364mwu12b47its9vkdcwnpqx9gndzfykpgyr&rid=giphy.gif"
    />
  </div>
</Fragment>
`;
